plenum/test/primary_selection/test_promotion_leads_to_primary_inconsistency.py

```diff
@@ -0,0 +1,83 @@
+import json
+
+from indy.did import create_and_store_my_did
+from plenum.test.node_catchup.helper import ensure_all_nodes_have_same_data
+
+from plenum.test.node_catchup.test_config_ledger import start_stopped_node
+
+from plenum.test.helper import sdk_send_random_and_check, checkViewNoForNodes
+from plenum.test.pool_transactions.helper import demote_node, disconnect_node_and_ensure_disconnected, promote_node
+from plenum.test.test_node import ensureElectionsDone, checkNodesConnected
+
+nodeCount = 7
+
+
+def test_promotion_leads_to_primary_inconsistency(looper,
+                                                  txnPoolNodeSet,
+                                                  tdir,
+                                                  tconf,
+                                                  allPluginsPath,
+                                                  sdk_wallet_steward,
+                                                  sdk_wallet_handle,
+                                                  sdk_pool_handle):
+    # We are saving pool state at moment of last view_change to send it
+    # to newly connected nodes so they could restore primaries basing on this node set.
+    # When current primaries getting edited because of promotion/demotion we don't take this into account.
+    # That lead us to primary inconsistency on different nodes
+
+    sdk_send_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle, sdk_wallet_steward, 1)
+    assert txnPoolNodeSet[0].master_replica.isPrimary
+    assert txnPoolNodeSet[1].replicas._replicas[1].isPrimary
+    assert txnPoolNodeSet[2].replicas._replicas[2].isPrimary
+    starting_view_number = checkViewNoForNodes(txnPoolNodeSet)
+
+    # Demote node 3
+    node_3 = txnPoolNodeSet[2]
+    steward_3_seed = 'Steward3' + '0' * 24
```

Comment:
We have `sdk_wallet_stewards` fixture to get each steward's did and wallet

Reply:
fixed